reconcile/test/conftest.py

```diff
@@ -36,3 +45,57 @@ def vault_secret():
         format=None,
         version=None,
     )
+
+
+def data_default_none(
+    klass: type[BaseModel], data: MutableMapping[str, Any]
+) -> MutableMapping[str, Any]:
+    """Set default values to None for required but optional fields."""
+    for field in klass.__fields__.values():
+        if not field.required:
+            continue
+
+        if field.alias not in data:
+            if field.allow_none:
+                data[field.alias] = None
```

Comment:
From a schemas perspective, all `Optional` fields are non-required, but everything else is required and must be set. Choosing an "empty value" can be dangerous, at least for `str`, `bool`, and `int`.

Let's keep it for the moment just for `Optional` values and maybe extend it in the future.

Reply:
Fully agree that this shouldn't be used in prod code, but for tests it might come in handy. But I agree that this could also be added later if deemed helpful.